src/Access/AccessControl.cpp

```diff
@@ -282,6 +282,8 @@ void AccessControl::setUpFromMainConfig(const Poco::Util::AbstractConfiguration
     setSettingsConstraintsReplacePrevious(config_.getBool("access_control_improvements.settings_constraints_replace_previous", false));
 
     addStoragesFromMainConfig(config_, config_path_, get_zookeeper_function_);
+
+    role_cache = std::make_unique<RoleCache>(*this, config_.getInt("access_control_improvements.role_cache_expiration_time", 600));
```

Comment:
Doesn't look very reliable if `setUpFromMainConfig()` will be somehow called more than once. But for now I suppose it's ok.

Reply:
Agree.
But as far as I can see setUpFromMainConfig() is not meant to be called more than once anyway.